src/query/stream/streams.cpp

```diff
@@ -157,37 +158,238 @@ void from_json(const nlohmann::json &data, StreamStatus<TStream> &status) {
 }
 
+namespace {
+template <typename Fun>
+[[nodiscard]] bool TryOrSetError(Fun &&func, mgp_result *result) {
+  if (const auto err = func(); err == MGP_ERROR_UNABLE_TO_ALLOCATE) {
+    static_cast<void>(mgp_result_set_error_msg(result, "Not enough memory!"));
+    return false;
+  } else if (err != MGP_ERROR_NO_ERROR) {
+    static_cast<void>(mgp_result_set_error_msg(result, "Unexpected error"));
+    return false;
+  }
+  return true;
+}
+
+auto GetStringValueOrSetError(const char *string, mgp_memory *memory, mgp_result *result) {
+  procedure::MgpUniquePtr<mgp_value> value{nullptr, mgp_value_destroy};
+  const auto success =
+      TryOrSetError([&] { return procedure::CreateMgpObject(value, mgp_value_make_string, string, memory); }, result);
+  if (!success) {
+    value.reset();
+  }
+
+  return value;
+}
+}  // namespace
+
 Streams::Streams(InterpreterContext *interpreter_context, std::filesystem::path directory)
     : interpreter_context_(interpreter_context), storage_(std::move(directory)) {
-  constexpr std::string_view proc_name = "kafka_set_stream_offset";
-  auto set_stream_offset = [this, proc_name](mgp_list *args, mgp_graph * /*graph*/, mgp_result *result,
-                                             mgp_memory * /*memory*/) {
-    auto *arg_stream_name = procedure::Call<mgp_value *>(mgp_list_at, args, 0);
-    const auto *stream_name = procedure::Call<const char *>(mgp_value_get_string, arg_stream_name);
-    auto *arg_offset = procedure::Call<mgp_value *>(mgp_list_at, args, 1);
-    const auto offset = procedure::Call<int64_t>(mgp_value_get_int, arg_offset);
-    auto lock_ptr = streams_.Lock();
-    auto it = GetStream(*lock_ptr, std::string(stream_name));
-    std::visit(utils::Overloaded{
-                   [&](StreamData<KafkaStream> &kafka_stream) {
-                     auto stream_source_ptr = kafka_stream.stream_source->Lock();
-                     const auto error = stream_source_ptr->SetStreamOffset(offset);
-                     if (error.HasError()) {
-                       MG_ASSERT(mgp_result_set_error_msg(result, error.GetError().c_str()) == MGP_ERROR_NO_ERROR,
-                                 "Unable to set procedure error message of procedure: {}", proc_name);
-                     }
-                   },
-                   [proc_name](auto && /*other*/) {
-                     throw QueryRuntimeException("'{}' can be only used for Kafka stream sources", proc_name);
-                   }},
-               it->second);
-  };
+  RegisterProcedures();
+}
 
-  mgp_proc proc(proc_name, set_stream_offset, utils::NewDeleteResource(), false);
-  MG_ASSERT(mgp_proc_add_arg(&proc, "stream_name", procedure::Call<mgp_type *>(mgp_type_string)) == MGP_ERROR_NO_ERROR);
-  MG_ASSERT(mgp_proc_add_arg(&proc, "offset", procedure::Call<mgp_type *>(mgp_type_int)) == MGP_ERROR_NO_ERROR);
+void Streams::RegisterProcedures() {
+  {
+    constexpr std::string_view proc_name = "kafka_set_stream_offset";
```

Comment:
Does that mean the procedure will be called like this `mg.kafka_set_stream_offset()`?

Reply:
yes